apps/settings/js/panels/homescreens/homescreen_name.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * HomescreenName module has a `name` property that contains the name of the
+ * currently used home screen and can be watched.
+ *
+ * @module HomescreenName
+ */
+define(require => {
+  'use strict';
+
+  const MANIFEST_URL_PREF = 'homescreen.manifestURL';
+
+  var SettingsListener = require('shared/settings_listener');
+  var AppsCache = require('modules/apps_cache');
+  var ManifestHelper = require('shared/manifest_helper');
+  var Module = require('modules/base/module');
+  var Observable = require('modules/mvvm/observable');
+
+  var HomescreenName = Module.create(function HomescreenName() {
+    this.super(Observable).call(this);
+    this._init();
+  }).extend(Observable);
+
+  /**
+   * Name of the currently installed home screen.
+   *
+   * @memberOf HomescreenName
+   * @type {String}
+   * @public
+   */
+  Observable.defineObservableProperty(HomescreenName.prototype, 'name', {
+    readonly: true,
+    value: ''
+  });
+
+  /**
+   * Init HomescreenName module.
+   *
+   * @private
+   */
+  HomescreenName.prototype._init = function hn_init() {
+    //this._name = '';
+    this._watchNameChange();
+  };
+
+  /**
+   * Watch the value of MANIFEST_URL_PREF from settings and update name
+   * accordingly.
+   *
+   * @private
+   */
+  HomescreenName.prototype._watchNameChange = function hn_watchNameChange() {
+    SettingsListener.observe(MANIFEST_URL_PREF, '', manifestURL => {
+      this._updateManifestName(manifestURL)
+        .then(name => {
```

Comment:
please put in the same line as `this._updateManifestName(manifestURL).then(name => {`